design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+
+Let's look at a specific example of the GCP cluster stack to understand the security and isolation model more thoroughly.
+First, the user will create an instance of a `ClusterStackInstall` that specifies which cluster stack they want to install.
+Then, when the Stack Manager installs the GCP stack, a few things will happen:
+
+* All CRDs related to GCP infrastructure and services will be installed to the cluster, such as `CloudSQLInstance`, `CloudSQLInstanceClass`, `GKECluster`, `GKEClusterClass`, `Network`, `Subnetwork`, etc.
+* A `ClusterRole` will be created at the cluster level that includes the RBAC rules for the [allowed core Kubernetes kinds](#allowed-resource-access), the stack's owned CRDs, and its dependent CRDs.
+* A `ServiceAccount` is created in the namespace that the stack's controller will be running.  This will serve as the runtime identity of the controller.
+* A `ClusterRoleBinding` is created at the cluster level to bind the `ClusterRole` and the namespaced `ServiceAccount`
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the GCP controllers.
+
+Because the stack is given a `ClusterRole` and `ClusterRoleBinding`, its permissions to the allowed resources are across all namespaces.
+
+### Architecture Diagram - Cluster Stack
+
+The diagram below summarizes the artifacts in the control plane after the GCP stack has been installed and a couple example `MySQLInstance` claims have been created.
+
+![Diagram of a Cluster Stack Installed](./images/cluster-stack-installed.png)
+
+### Environments
+
+While cluster stacks are installed at the cluster level (there is only one instance of each cluster stack), they can still support the concept of multiple "environments", each represented by a namespace.
+For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
+All resources within each of those environments can share a common set of provider credentials.
+This namespaced "environment" approach provides the administrator with a way to collect or aggregate all the infrastructure and services associated with a given environment.
+
+#### Cluster Stack Single Instance Version Limitation
+
+One caveat of this approach is that because there is only a single instance of the cluster stack in the cluster, it is therefore shared across all environments.
+This means that all environments must use the same version of the stack.
+If an admin wants to try a newer version of the stack for their dev environment, it will also result in the newer version of the stack being used in the production environment as well.
+
+This limitation could be addressed in the future by allowing multiple versions of cluster stacks installed per control plane, which would require additional complexity and design work.
```

Comment:
In this model then, we are saying that if someone wants to connect to multiple AWS accounts (for example), they should run multiple different Crossplane clusters?

Related question: do we need to support multiple versions of a stack in a single Crossplane cluster, if someone can easily spin up a new cluster?

Reply:
If someone wants to use multiple AWS accounts, I'd expect them to create multiple `provider.aws.crossplane.io` objects within the cluster.  The single AWS stack can watch for all of these.

There shouldn't be a reason to create multiple Crossplane clusters in order to use multiple AWS accounts.

Reply:
I don't think we want to commit to not supporting multiple versions of a stack in the future quite yet.  I would personally view having to spin up another cluster/control-plane and add that to my list of environments to worry about as a negative experience.